ambari-server/src/main/java/org/apache/ambari/server/topology/DownloadMpacksTask.java

```diff
@@ -72,9 +73,12 @@ public void downloadMissingMpacks(Collection<MpackInstance> mpackInstances) {
         }
       }
       catch (Exception ex) {
-        throw ex instanceof RuntimeException ?
-          (RuntimeException)ex :
-          new RuntimeException("Error occured while registering mpack: " + mpack.getStackId(), ex);
+        List<Throwable> causes = ExceptionUtils.getThrowableList(ex);
+        Throwable rootCause = causes.get(causes.size() - 1);
```

Comment:
Can the list of Throwables ever be empty?  

If so, then the same NullPointerException could occur when the "rootCause" variable is accessed a few lines down from here.  

From the javadoc for this method, it looks like at least the input Throwable is always returned, but it would be good to confirm this.  

If this method always returns at least one Exception, then this probably isn't a problem, but I thought it made sense to double-check this.

Reply:
The doc is right, at least the input exception is returned.